src/wiener.jl

```diff
@@ -52,8 +52,14 @@ function wiener(input::AbstractArray, signal::AbstractArray,
 end
 
 # Noise as a real (it's converted to an array)
-wiener(input::AbstractArray, signal::AbstractArray, noise::Real) =
-    wiener(input, signal, fill!(similar(input), one(eltype(input)))*noise)
+function wiener(input::AbstractArray, signal::AbstractArray, noise::Real)
+    @assert size(input) == size(signal)
+    input_ft = fft(input)
```

Comment:
I have some suggestion and I have no strong opinion about this refactoring. There is some code repetition. I would trade some performance for readability.

```julia
function whitenoise(noiselevel::Real, size)
    result = zeros(size)
    result[1] = noiselevel
    return result
end

function wiener(input::AbstractArray, signal::AbstractArray, noiselevel::Real)
    return wiener(input, signal, whitenoise(noiselevel, size(input)))
end

function wiener(input::AbstractArray, signal::AbstractArray,
                noiselevel::AbstractArray, blurring::AbstractArray)
    return wiener(input, signal, whitenoise(noiselevel, size(input)), blurring)
end
```

Reply:
Yeah, I was also not sure how to handle this repetition. 

You mean as performance drop this extra fft call on the whitenoise array? 
Furthermore I was wondering if we should rewrite the wiener function itself because we can also save the whitenoise array completely and instead just add this noise value point wise in the denominator.

```julia
return real(ifft(Y .* conj(H) ./ (abs2.(H) .+ noise ./ S)))
```


Btw, as you might have seen I'm also updating the documentation. I recognized that the example in the documentation is outdated but working code is in `time-series.jl`. So I transferred the code from this file to the documentation.
Also I would like to add an example of a deconvolution of a time series with the Wiener filter.

Reply:
> You mean as performance drop this extra fft call on the whitenoise array?

Yes, that was my concern.


> Furthermore I was wondering if we should rewrite the wiener function itself because we can also save the whitenoise array completely and instead just add this noise value point wise in the denominator.

I am not sure if white noise is so common to optimise for this case. If you and @giordano agree, let's do that.


> Btw, as you might have seen I'm also updating the documentation

Thanks!


> Also I would like to add an example of a deconvolution of a time series with the Wiener filter.

This is awesome, but I think some kind of STFT should be used for that. I would like to see such example :)

Reply:
I pushed the noisy time series and also added a example of using the constant noise parameter.

In this case it doesn't work so well because the image is affected heavily by noise. With less noise and more blurring, the constant noise parameter is actually more powerful.

Reply:
Docs look good now, let's ignore my comment and after making tests pass I will merge